src/main/java/org/jboss/logmanager/formatters/TextBannerFormatter.java

```diff
@@ -37,6 +37,9 @@ public TextBannerFormatter(final Supplier<String> bannerSupplier, final ExtForma
     public String getHead(final Handler h) {
         final String dh = Objects.requireNonNullElse(delegate.getHead(h), "");
         final String banner = Objects.requireNonNullElse(bannerSupplier.get(), "");
+        if (dh.isEmpty()) {
```

Comment:
This check already exists in `String.concat` does it not?

Reply:
Nope - AFAIK the indy got to be created, regardless and got hot

Reply:
Ah, maybe the better fix is to change the `+` to `String.concat` then. For two simple strings, an indy is unlikely to do better.

Reply:
lovely, good idea! https://github.com/openjdk/jdk/blob/752e1629555f0ec8630373ec87b049afdd709ea6/src/java.base/share/classes/java/lang/String.java#L2988